core/store/src/trie/mod.rs

```diff
@@ -756,15 +757,14 @@ impl Trie {
     /// # Errors
     ///
     /// If the root or any subsequent nodes are not found in the trie, `StorageError` is returned
-    pub fn get_proof(
-        &self,
-        root: &CryptoHash,
-        key: &[u8],
-    ) -> Result<(ProofState, Vec<RawTrieNodeWithSize>), StorageError> {
+    pub fn get_proof(&self, root: &CryptoHash, key: &[u8]) -> Result<Arc<Vec<u8>>, StorageError> {
```

Comment:
I meant `Result<(ProofState, Vec<Arc<[u8]>>), StorageError>` and then you can just do `levels.push(bytes)` after decoding the node.